base_local_planner/src/simple_trajectory_generator.cpp

```diff
@@ -131,6 +137,15 @@ void SimpleTrajectoryGenerator::initialise(
         vel_samp[1] = y_it.getVelocity();
         for(; !th_it.isFinished(); th_it++) {
           vel_samp[2] = th_it.getVelocity();
+
+          // spin in place if turn radius less than 10cm
+          if (hypot(vel_samp[0], vel_samp[1]) / abs(vel_samp[2]) < 0.1) {
+            const double t = (vel_samp[2] - min_vel[2]) / (max_vel[2] - min_vel[2]);
+            vel_samp[0] = 0;
+            vel_samp[1] = 0;
+            vel_samp[2] = min_vel[3] + t * (max_vel[3] - min_vel[3]);
+          }
```

Comment:
hmm I don't really like the 0.1 number for the same reasons as in the initial comment.

What I would do is just at the end of the function, do:
```
VelocityIterator th_spin_it(min_vel[3], max_vel[3], vsamples[2]);
for(; !th_spin_it.isFinished(); th_spin_it++) {
    if (th_spin_it.getVelocity() < min_vel[2] || th_spin_it.getVelocity() > max_vel[2]) {
        sample_params_.push_back({0, 0, th_spin_it.getVelocity()});
   }
}
```

and if we're worried about "cheating the user" by adding more samples, I'd just leave a comment at the `min_vel_th_spin` parameter that setting it larger than `min_vel_th` will insert at max `vth_samples` new samples for spinning in place

Reply:
> hmm I don't really like the 0.1 number for the same reasons as in the initial comment.

What reason? This one? _depending on the vsamples / acc_lim params and the 0.05 value here, there is maybe no spin_in_place among the candidates._

That's an entirely different problem. I think you can agree with me that if a trajectory has a turn radius higher than 10cm (or some other arbitrary threshold), then it is no longer a spin turn, correct? The problem you're talking about is whether or not DWA will even generate such trajectories -- that is not something we are trying to fix in this PR.

The purpose of this PR is to simply increase the spin in place velocity limit, not make it happen more frequently.

Reply:
to sum up:
- If we inject trajectories, I'd do what I wrote above
- If we want to modify the generated trajectories, we shouldn't modify x/y values; what I would do is just stretch the th samples over a larger interval, sth like:
```
    VelocityIterator x_it(min_vel[0], max_vel[0], vsamples[0]);
    VelocityIterator y_it(min_vel[1], max_vel[1], vsamples[1]);
    for(; !x_it.isFinished(); x_it++) {
      vel_samp[0] = x_it.getVelocity();
      for(; !y_it.isFinished(); y_it++) {
        vel_samp[1] = y_it.getVelocity();
        const auto v_mag = hypot(vel_samp[0], vel_samp[1]);
        auto th_it = v_mag > 2 * limits->min_vel_trans ? VelocityIterator(min_vel[2], max_vel[2], vsamples[2]) : VelocityIterator(min_vel[3], max_vel[3], vsamples[2]);
        for(; !th_it.isFinished(); th_it++) {
          vel_samp[2] = th_it.getVelocity();
          //ROS_DEBUG("Sample %f, %f, %f", vel_samp[0], vel_samp[1], vel_samp[2]);
          sample_params_.push_back(vel_samp);
        }
      }
      y_it.reset();
    }
  }
```